Source/WebGPU/WGSL/ConstantValue.h

```diff
@@ -36,22 +36,38 @@ namespace WGSL {
 #if HAVE(FP16_HALF_SUPPORT)
 using half = __fp16;
 #else
+// Wrap a struct around the supported fp16 type.
 struct half {
+#if PLATFORM(__APPLE__)
+    using f16 = __fp16;
+#else
+    // _Float16 is the 16bit float type in C++23, and is often available
+    // in compilers prior to C++23.
+    using f16 = _Float16;
+#endif
     half()
     {
     }
 
-    half(auto val)
-        : value(static_cast<float>(val))
-    {
-    }
+    // Constructor from an arithmetic type. Use a template here because the
```

Comment:
Are these changes required? I was originally trying to avoid implicit conversions as it makes it ambiguous when constructing a `ConstantValue`.

Reply:
My first attempt at the patch listed individual types like `double` `int` etc.  It worked on GCC x86-64 but failed compilation on the Mac platforms with Clang because I had misted `long long int`. So I switched it to a template based on arithmetic types.

You can see the first attempt on the left here: https://github.com/WebKit/WebKit/compare/e99f5a6caf0e382f565e1d53886e680e75a1159c..b5e47f1991101cb3eb15466d30ee57a0c0271046#diff-4daad8a03b2ec07ff1020c4a221a2ae9c81e47f11c46d407f8c76d56f486844f

Reply:
Hum, not sure about the other errors, looks like it's due to an unfortunate implicit cast. It's existing code though, and it's nice that the code is a bit more explicit now, so I'm happy to land as is and I can look into the casts later